internal/pkg/manifest/workload_ecs.go

```diff
@@ -85,13 +85,26 @@ type ImageWithPortAndHealthcheck struct {
 	HealthCheck   ContainerHealthCheck `yaml:"healthcheck"`
 }
 
+// AlarmArgs represents specs of CloudWatch alarms for deployment rollbacks.
+type AlarmArgs struct {
+	CPUUtilization    *int `yaml:"cpu_utilization"`
+	MemoryUtilization *int `yaml:"memory_utilization"`
+}
+
+// RollbackAlarmArgsOrNames represents alarms for rolling back deployments.
+// It supports unmarshalling yaml which can either be of type AlarmArgs or a list of strings.
+type RollbackAlarmArgsOrNames struct {
+	Union[[]*string, AlarmArgs]
+}
+
 // DeploymentConfiguration represents the deployment strategies for a service.
 type DeploymentConfiguration struct {
-	Rolling *string `yaml:"rolling"`
+	Rolling *string                  `yaml:"rolling"`
+	Alarms  RollbackAlarmArgsOrNames `yaml:"rollback_alarms"`
```

Comment:
based on the usage - i don't think you need a separate struct here!
```suggestion
	Alarms  Union[[]*string, AlarmArgs] `yaml:"rollback_alarms"`
```

Reply:
also - i think the first type can just be `[]string`, not `[]*string`